src/NHibernate/Proxy/DynamicProxy/ProxyFactory.cs

```diff
@@ -118,13 +118,13 @@ private TypeInfo CreateUncachedProxyType(System.Type baseType, IReadOnlyCollecti
 			
 			// Provide a custom implementation of ISerializable
 			// instead of redirecting it back to the interceptor
-			foreach (MethodInfo method in ProxyBuilderHelper.GetProxiableMethods(baseType, interfaces).Where(method => method.DeclaringType != typeof(ISerializable)))
+			foreach (MethodInfo method in ProxyBuilderHelper.GetProxiableMethods(parentType, interfaces).Where(method => method.DeclaringType != typeof(ISerializable)))
 			{
 				ProxyMethodBuilder.CreateProxiedMethod(interceptorField, method, typeBuilder);
 			}
 
 			// Make the proxy serializable
-			AddSerializationSupport(baseType, baseInterfaces, typeBuilder, interceptorField, defaultConstructor);
+			AddSerializationSupport(parentType, baseInterfaces, typeBuilder, interceptorField, defaultConstructor);
```

Comment:
This one must stay `baseType`, otherwise when deserializing a proxy with a base type being an interface, it will create a new proxy type with `baseType` `DummyProxy` instead of the interface, causing proxy type caching issues. (Two types for the same proxy may be built.)